android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyTest.kt

```diff
@@ -0,0 +1,214 @@
+package org.greatfire.envoy
+
+import android.net.Uri
+import android.net.UrlQuerySanitizer
+import android.util.Log
+
+// old shadowsocks
+import android.content.Intent
+import androidx.core.content.ContextCompat
+
+// This class represents an Envoy connection: type and URL,
+// and tracks additional information, such as a URL to any
+// proxy used, e.g. Shadowsocks provides a SOCKS5 interface
+// which is what our Interceptor actually uses
+//
+// The url should be the one provided by the caller
+// testType lets us know how to treat that URL
+// some transports have an additional URL, e.g. a SOCKS5 URL
+// for a PT, that gets stored in proxyUrl
+// proxyiSEnvoy lets us know if proxyUrl is an Envoy proxy or SOCKS/HTTP
+
+data class EnvoyTest(
+    var testType: EnvoyServiceType,
+    var url: String,
+) {
+    companion object {
+        private const val TAG = "EnvoyTest"
+    }
+
+    // proxy URL for the service providing transport
+    // can be SOCKS5, HTTP(S), Envoy, see proxyIsEnvoy
+    var proxyUrl: String? = null
+    // - true means the proxyUrl refers to a nonstandard Envoy proxy
+    // as documented at http://github.com/greatfire/envoy/
+    // - false means it's SOCKS or HTTP
+    var proxyIsEnvoy: Boolean = false
+
+    // Is this the service we chose to use
+    var selectedService: Boolean = false
```

Comment:
since the revisions, it seems like this is only set and never read

Reply:
It's was just there to make sure we didn't stop the active service when I was stopping everything in a loop at the end... but that works differently now